### PR TITLE
[#171] Implemented Dex authentication in Robot tests

### LINKED ISSUE #171: Fix Dex auth module in Airflow to allow tests access Rest API
Currently "Deploy legion" jenkins job fails in Regression Testing stage, due to inaccessible Airflow secured by Dex.

### PR DESCRIPTION
**Pull request by aliaksandr-d**
Added Dex authentication support to Robot tests
Now robot tests take credentials of test user from profile (if Dex is enabled) and send requests to services using authorized session ID.
**This PR closes #171  but should be merged ONLY after PR #160 and PR #168**

### CODE REVIEW

File: legion_test/legion_test/robot/dex_client.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    :return: session ID value
+    """
+    if _session_id:
+        return _session_id
+    else:
+        raise ValueError('Session ID is not inited')
+
+
+def get_session_cookies() -> dict:
```
Comment (kirillmakhonin) [inline legion_test/legion_test/robot/dex_client.py]: We currently dont use typing for functions

File: legion_test/legion_test/robot/dex_client.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    """Get session cookies that can be used inside Request.
+    :return: cookies dict
+    """
+    if _session_id:
+        return {SESSION_ID_COOKIE_NAME: _session_id}
+    else:
+        # Dex is disabled or Session ID wasn't found
+        return
```
Comment (kirillmakhonin) [inline legion_test/legion_test/robot/dex_client.py]: Documentation conflicts with return type

File: tests/load_variables_from_profiles.py [hunk 3]
```diff
@@ -66,4 +67,9 @@ def get_variables(arg):
     variables['HOST_PROTOCOL'] = 'https' if variables['USE_HTTPS_FOR_TESTS'] else 'http'
     variables['MODEL_TEST_ENCLAVE'] = variables['ENCLAVES'][0] if len(variables['ENCLAVES']) > 0 else 'UNKNOWN_ENCLAVE'
 
+    if 'dex' in data and data['dex']['enabled'] and 'staticPasswords' in data['dex']['config'] and \
```
Comment (kirillmakhonin) [inline tests/load_variables_from_profiles.py]: We should move auth credentials to secrets

**Comment by aliaksandr-d** [comment]
Test this please

File: tests/load_variables_from_profiles.py (hunk 3 above)
Comment (aliaksandr-d) [reply tests/load_variables_from_profiles.py]: Creds are moved to secrets

**Comment by aliaksandr-d** [comment]
Moved to separate PR, where all DEX functionality is aggregated.

**Pull request closed without merging**